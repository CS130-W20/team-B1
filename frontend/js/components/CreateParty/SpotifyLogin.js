import React, { Component } from 'react';
import { Button } from 'reactstrap';
import { Route } from 'react-router';

import { loginWithSpotify } from '../../utils/api';

// import 'bootstrap/dist/css/bootstrap.css';
// import '../css/MainScreen.css';

class SpotifyLogin extends Component {
	constructor() {
    super();
    this.state = {
			redirect: false,
			oauthURL: null,
		};
  }

	handleLogin() {
		loginWithSpotify().then(json => {
			this.setState({oauthURL: json['location'], redirect: true})
		});
	}
	
	render () {
		if (this.state.redirect) {
			return <Route path='/login' component={() => { 
				window.location.href = this.state.oauthURL; 
				return null;
	 		}}/>;
		}

		return (
			<div className="SpotifyLogin">
				<div className="buttons">
					<Button className="button" color="success" onClick={this.handleLogin.bind(this)}> Login to Spotify
						<small> You are the life of the party! </small>
					</Button>
				</div>
			</div>
		);
	}
}

export default SpotifyLogin;
