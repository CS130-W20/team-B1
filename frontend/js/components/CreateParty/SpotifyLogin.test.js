import React from 'react';
import renderer from 'react-test-renderer';

import SpotifyLogin from './SpotifyLogin';

describe ('Testing Spotify Login Page', () => {
    it('correctly renders', () => {
        const wrapper = renderer.create(<SpotifyLogin />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
