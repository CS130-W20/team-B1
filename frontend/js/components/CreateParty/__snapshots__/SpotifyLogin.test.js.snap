// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Spotify Login Page correctly renders 1`] = `
<div
  className="SpotifyLogin"
>
  <div
    className="buttons"
  >
    <button
      aria-label={null}
      className="button btn btn-success"
      onClick={[Function]}
    >
       Login to Spotify
      <small>
         You are the life of the party! 
      </small>
    </button>
  </div>
</div>
`;
